internal/pkg/addons/cloudformation.go

```diff
@@ -0,0 +1,92 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package addons
+
+import (
+	"reflect"
+
+	"gopkg.in/yaml.v3"
+)
+
+type cfnSection int
+
+const (
+	metadataSection cfnSection = iota + 1
+)
+
+// cloudformation represents a parsed YAML AWS CloudFormation template.
```

Comment:
What do you think of renaming to "cfnTemplate" which will be more informative and avoiding struct and pkg sharing the same name `cloudformation`?

Reply:
Done!